mitodl/mit-open
# user list UI updates

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4789

### Description (What does it do?)
This PR updates the Dashboard "My Lists" section to be in line with the current designs, as noted in the issue. The component was also updated to use a more standard routing structure, rather than a hash to keep track of the current tab:

 - Dashboard home: `/dashboard/`
 - My lists: `/dashboard/my-lists/`
 - Individual list: `/dashboard/my-lists/1`
 - Profile: `/dashboard/profile/`

### Screenshots (if appropriate):
Note: These were taken before a rebase added the new Settings tab
![image](https://github.com/user-attachments/assets/e252edca-0ab5-4273-8dc4-3ff4035d689d)
![image](https://github.com/user-attachments/assets/a82ea746-9c5e-4ec1-b7eb-97fd6f341a89)
![image](https://github.com/user-attachments/assets/67045de5-ba20-4ff6-b7d8-f6636583ad80)
![image](https://github.com/user-attachments/assets/5c267dfd-d0d1-4fe8-8ac3-a012cd3ed9a1)
![image](https://github.com/user-attachments/assets/c165f450-9af1-401a-b868-6f13e52df682)

### How can this be tested?
 - Spin up `mit-open` on this branch, making sure you have at least some courses backpopulated into your instance and you have a user to log in with
 - Go to http://localhost:8063/dashboard
 - Click around on the different tabs and ensure the pages load properly
 - Click on the "My Lists" tab and create a list if you don't already have one
 - Click on Home
 - Add some courses to your list by clicking the bookmark icons on any of the cards
 - Go back to My Lists
 - Click on the list
 - Click the rerorder button and drag the items around
 - Throughout this process, ensure that what you see reflects the designs
